### PR TITLE
Added ability to EXPLAIN queries in Debug panel

### LINKED ISSUE #74: does yii\mongo\query support explain and hint method ?
This issue has originally been reported by @monochrome-yeh at https://github.com/yiisoft/yii2/issues/9413.
Moved here by @cebe.

-----

native method
http://docs.mongodb.org/manual/reference/operator/meta/explain/
http://docs.mongodb.org/manual/reference/operator/meta/hint/

how can i use ActiveRecord or ActiveQuery for these method?

### PR DESCRIPTION
**Pull request by webdevsega**
I add explain queries in debug panel and method in Query class.

fixes #74

### CODE REVIEW

**Comment by alexandernst** [comment]
Any news on this? Is this reviewed? Ready to be merged? Something missing?

File: debug/ExplainAction.php [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+        $this->controller->loadData($tag);
+
+        $timings = $this->panel->calculateTimings();
+
+        if (!isset($timings[$seq])) {
+            throw new HttpException(404, 'Log message not found.');
+        }
+
+        $query = $timings[$seq]['info'];
+        preg_match('/^.+\((.*)\)$/', $query, $matches);
```
Comment (samdark) [inline debug/ExplainAction.php]: Could be

```php
if (!preg_match('/^.+\(.+\)$/', $query, $matches)) {
    return;
}
```

File: debug/MongoDbPanel.php [hunk 2]
```diff
@@ -48,4 +59,32 @@ public function getProfileLogs()
...
+        return true;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    protected function getQueryType($timing)
+    {
+        $timing = ltrim($timing);
+        $timing = mb_substr($timing, 0, mb_strpos($timing, '('));
```
Comment (samdark) [inline debug/MongoDbPanel.php]: Encoding should be specified.

File: debug/MongoDbPanel.php [hunk 3]
```diff
@@ -48,4 +59,32 @@ public function getProfileLogs()
...
+
+        return count($matches) ? array_pop($matches) : '';
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public static function canBeExplained($type)
+    {
+        return $type == 'find';
```
Comment (samdark) [inline debug/MongoDbPanel.php]: `===` could be used.

File: debug/MongoDbPanel.php (hunk 2 above)
Comment (webdevsega) [inline debug/MongoDbPanel.php]: Right? mb_substr($timing, 0, mb_strpos($timing, '('), 'utf8')

File: debug/ExplainAction.php (hunk 1 above)
Comment (webdevsega) [inline debug/ExplainAction.php]: @samdark I need to get a substring in parentheses, so your version does not suit you.

**Comment by klimov-paul** [comment]
I am unsure this will work, since actual query is not written in the logs.
Logs contain only a close approximation to the actual query been performed.
Although it may work in some cases, explain may fail unexetecly for some queries.

**Comment by webdevsega** [comment]
@klimov-paul This works only for FIND queries. Now all FIND queries are logged using MongoCursor::info, so everything is standardized and should be no problem.

File: debug/ExplainAction.php (hunk 1 above)
Comment (samdark) [inline debug/ExplainAction.php]: Right. My mistake.

File: debug/MongoDbPanel.php (hunk 2 above)
Comment (samdark) [inline debug/MongoDbPanel.php]: Yes.

**Comment by klimov-paul** [comment]
Merged manually.
Thank you.

**Pull request merged**